instance/models/openedx_instance.py

```diff
@@ -389,3 +394,117 @@ def delete(self, *args, **kwargs):
         self.deprovision_s3()
         self.deprovision_rabbitmq()
         super().delete(*args, **kwargs)
+
+    @staticmethod
+    def compare(first, second, encoding='utf8'):
+        """
+        Compares mixed values of bytes and string objects to
+        check whether the contain the same value.
+
+        :param first: The first argument to compare
+        :param second: The second argument to compare
+        :param encoding: The bytes' objects expected encoding.
+        :return: Whether the value of the first argument equals the second's
+        """
+        if isinstance(first, bytes):
+            first = first.decode(encoding)
+        if isinstance(second, bytes):
+            second = second.decode(encoding)
+
+        return first == second
+
+    @property
+    def consul_prefix(self):
+        """
+        This is a property that helps determining a specific instance's rpefix in
+        Consul. The prefix helps in keeping different instances' configurations
+        separate from overridings and modifications.
+
+        :return: A unique Key-Value prefix for this instance in Consul.
+        """
+        return '{ocim}/instances/{instance}/'.format(ocim=settings.OCIM_ID, instance=self.pk)
+
+    def _generate_consul_configurations(self):
+        """
+        Collects required configurations for this instance to reflect on Consul.
+        You can add, delete or alter configurations keys and values here which are
+        going to ve reflected in Consul immediately.
+
+        :return: A dict of the configurations.
+        """
+        active_servers = self.get_active_appservers()
+        active_servers_names = [server.name for server in active_servers]
```

Comment:
The information we need for each app server is the id (`appserver.pk`) and the public IP address (`appserver.server.public_ip`).  We eventually want to build a load balancer configuration from this information.  See [this code](https://github.com/open-craft/opencraft/blob/d5393bcda6fb26395357efb7ebbfc01eddf1bce5/instance/models/openedx_instance.py#L119-L131) for additional context.  I think a list of dictionaries would work here.

As a performance note, the current code will query the database for all appserver fields, build an `OpenEdXAppServer` instance for each of them and then extracts the `name` field.  A more efficient solution that only retrieves the required fields from the database is to use the `values()` methos, e.g.

    active_server_data = active_servers.values('pk', 'server__public_ip')

(I didn't verify the result for this particular example, but in general this results in close to optimal SQL queries, without making the code less readable.)

Reply:
Another side note:  We are trying to keep the information in Consul in sync with the information in the database.  Both the id and the IP address of an app server don't ever change, so we don't need any provisions to keep them in sync; we only need to update the list if an app server gets activated or deactivated, so the code we have should be enough.